src/VisualStudio/Core/Def/RoslynPackage.cs

```diff
@@ -127,6 +121,13 @@ private void LoadComponentsBackground()
 
             this.ComponentModel.GetService<MiscellaneousTodoListTable>();
             this.ComponentModel.GetService<MiscellaneousDiagnosticListTable>();
+
+            // Initialize any experiments async
+            var experiments = this.ComponentModel.DefaultExportProvider.GetExports<IExperiment>();
+            foreach (var experiment in experiments)
+            {
+                await experiment.Value.Initialize().ConfigureAwait(false);
```

Comment:
So the presence of a `ConfigureAwait(false)` would lead me to believe that calling `Initialize` has no threading requirements but `KeybindingResetDetector` asserts that it is must be on the UI thread.  This code works fine today as there is only one `IExperiment` type to call `Initialize` on but if there are ever multiple types that implement `IExperiment` then this could fail as we resume onto a non-UI thread after calling `Initialize`

Reply:
Hm? `KeybindingResetDetector.Initialize` has no threading requirement. It kicks off `InitializeCore` which is called on the UI thread below input priority.

Reply:
sorry, my brain skipped over the call to `ForegroundThreadAffinitizedObject.InvokeBelowInputPriority`.  This is obviously fine.